### PR TITLE
add : product page and filter functionality

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
### Files to check:
- ProductFilterBar.jsx
- Products.jsx
- getFilterData.jsx

### Files to ignore:
- CSS files
- database files

### Live link: https://dazzle-cerashop-4q92e8tvt-apurvasawant11.vercel.app/

Note: Click on `Shop All` button in header to navigate to the products page or add route `/products` in URL

### CODE REVIEW

File: src/context/data-context.jsx [hunk 1]
```diff
@@ -3,41 +3,42 @@ import { initialReducerData, dataReducer } from "../reducer/dataReducer";
...
+          payload: productsList.products,
         });
       } catch (error) {
-        dispatch({
-          type: "ERROR",
-          payload: "Error: Something is Wrong",
-        });
+        console.log(error);
       }
     })();
```
Comment (vrinda-mahajan) [inline src/context/data-context.jsx:30]: Appreciation:- Liked the use of iife function here.

File: src/context/data-context.jsx [hunk 2]
```diff
@@ -3,41 +3,42 @@ import { initialReducerData, dataReducer } from "../reducer/dataReducer";
...
-          payload: ProductsResponse.data.products,
+          type: "INITIALIZE_PRODUCTS",
+          payload: productsList.products,
         });
       } catch (error) {
-        dispatch({
-          type: "ERROR",
-          payload: "Error: Something is Wrong",
-        });
+        console.log(error);
```
Comment (vrinda-mahajan) [inline src/context/data-context.jsx:28]: Suggestion- You can use `console.error` instead of `console.log` here.

**Review by vrinda-mahajan — APPROVED** [review]
Your code looks clean and readable. You can merge it.

File: src/pages/product/ProductFilterBar.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            min="0"
+            max="19999"
+            value={priceRange.max}
+            onChange={(e) => onPriceChangeHandler("max", e.target.value, e)}
+          />
+          <div className="slider">
+            <div
+              className="progress"
+              style={{
+                left: `${sliderValue.left}%`,
```
Comment (abhisheksahani-nova) [inline src/pages/product/ProductFilterBar.jsx:87]: Try to avoid inline styles

**Review by abhisheksahani-nova — APPROVED** [review]
UI looks good .

Suggestion :- Can work on making code even better readable and simplified, else great work

File: src/context/data-context.jsx (hunk 2 above)
Comment (abhisheksahani-nova) [reply src/context/data-context.jsx:28]: One more thing , its better to remove console from code before pushing to remote

File: src/context/data-context.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/context/data-context.jsx:30]: Thanks!

File: src/context/data-context.jsx (hunk 2 above)
Comment (ApurvaSawant11) [reply src/context/data-context.jsx:28]: Updated `console.log` to `console.error`.

File: src/pages/product/ProductFilterBar.jsx (hunk 3 above)
Comment (ApurvaSawant11) [reply src/pages/product/ProductFilterBar.jsx:87]: I am updating the values dynamically based on user input. Hence, using inline styles here.

**Comment by ApurvaSawant11** [comment]
> UI looks good .
> 
> Suggestion :- Can work on making code even better readable and simplified, else great work

I will work on that. Thanks for reviewing

**Pull request merged**